e2etest/newe2e_task_runazcopy.go

```diff
@@ -277,6 +277,7 @@ func RunAzCopy(a ScenarioAsserter, commandSpec AzCopyCommand) (AzCopyStdout, *Az
 	}
 
 	err = command.Wait()
+	a.Assert(out.String(), common.Iff[Assertion](commandSpec.ShouldFail, Not{IsNil{}}, IsNil{}), err)
```

Comment:
This is unnecessary in the final PR. My plan file testing PR includes changes that output the standard output as a part of the log output on failures.